pkg/provider/browser/browser.go

```diff
@@ -56,10 +72,37 @@ func (cl *Client) Authenticate(loginDetails *creds.LoginDetails) (string, error)
 		Headless: playwright.Bool(cl.Headless),
 	}
 
-	// currently using Chromium as it is widely supported for Identity providers
+	validBrowserTypes := []string{"chromium", "firefox", "webkit", "chrome", "chrome-beta", "chrome-dev", "chrome-canary", "msedge", "msedge-beta", "msedge-dev", "msedge-canary"}
```

Comment:
why differentiate all these `browsertypes` if playwright only supports firefox, chrome and webkit?

Reply:
Hi @gliptak, Thank you for your review.

Playwright does support all of those browser types, see:
https://playwright.dev/docs/browsers and https://github.com/playwright-community/playwright-go/blob/main/generated-structs.go#L691C4-L695

The main idea behind this PR is that the user can choose a browser type which is already installed on their system.

If you choose a browser you do not have on your system, you will get a relevant error message.

For example, I tried to set `--browser-type=msedge`. Since I did not have `msedge` installed I got the following notification:  

```
INFO[0000] Setting browser type: msedge                  provider=browser
Error authenticating to IdP.: could not send message: Chromium distribution 'msedge' is not found at /Applications/Microsoft Edge.app/Contents/MacOS/Microsoft Edge
Run "npx playwright install msedge"
```
After I installed `msedge`, it used the installed one:
```
INFO[0000] Setting browser type: msedge                  provider=browser
INFO[0001] opening browser                               URL="[IDP_URL]" provider=browser
INFO[0021] waiting ...                                   provider=browser
INFO[0049] clean up browser                              provider=browser
```

Furthermore, If your browser is installed on a non-default location, you can utilize `--browser-executable-path`, and it will start the browser from that path.

Hope this explains it.